src/weather/weather.controller.ts

```diff
@@ -0,0 +1,16 @@
+import { Controller, Get, Query } from '@nestjs/common';
+import { WeatherService } from './weather.service';
+import { WeatherApiResponse } from './models/weather.model';
+import { GetForecastQuery } from './input/get-forecast.query';
+
+@Controller('weather')
+export class WeatherController {
+  constructor(private readonly weatherService: WeatherService) {}
+
+  @Get()
+  async getWeather(
+    @Query() query: GetForecastQuery,
+  ): Promise<WeatherApiResponse | { error: string }> {
+    return await this.weatherService.getForecast(query.lat, query.lng);
+  }
```

Comment:
While I like the usage of error as value, we usually make the return type the success path and the error path is described in the documentation with use of the decorators, e.g.

```typescript
  @Get()
  @ApiBadRequestResponse({
    description: 'Validation error',
    // either write schema yourself
    schema: { type: 'object', properties: { message: { type: 'string' } } },
    // or pass a type (constructor to a error class with decorated properties)
    type: WeatherError,
  })
  async getWeather(
    @Query() query: GetForecastQuery,
  ): Promise<WeatherApiResponse> {
    return await this.weatherService.getForecast(query.lat, query.lng);
  }
```